File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/mutability/ValAssignedInIfStatement.java
// IGNORE_K2

import org.jetbrains.annotations.*;

class A {

    public String stringAssignment() {
        if (Math.random() > 0.50) {
            return "a string";
        }
        return "b string";
    }

    public void aVoid() {
        String aString;
        String bString = stringAssignment();
        String cString = null;
        if (bString.startsWith("b")) {
            aString = "bbbb";
            cString = "cccc";
        } else {
            aString = stringAssignment();
            cString = stringAssignment();
        }
    }
}

File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/mutability/ValAssignedInIfStatement.kt
internal class A {
    fun stringAssignment(): String {
        if (Math.random() > 0.50) {
            return "a string"
        }
        return "b string"
    }

    fun aVoid() {
        val aString: String
        val bString = stringAssignment()
        var cString: String? = null
        if (bString.startsWith("b")) {
            aString = "bbbb"
            cString = "cccc"
        } else {
            aString = stringAssignment()
            cString = stringAssignment()
        }
    }
}
